src/sentry/tasks/post_process.py

```diff
@@ -333,49 +341,33 @@ def post_process_group(
 
         _capture_stats(event, is_new)
 
-        from sentry.models import GroupInboxReason
-        from sentry.models.groupinbox import add_group_to_inbox
-
-        with sentry_sdk.start_span(op="tasks.post_process_group.add_group_to_inbox"):
-            try:
-                if is_reprocessed and is_new:
-                    add_group_to_inbox(event.group, GroupInboxReason.REPROCESSED)
-            except Exception:
-                logger.exception("Failed to add group to inbox for reprocessed groups")
+        if not is_reprocessed or is_new:
+            has_reappeared = not is_new
 
-        if not is_reprocessed:
             # we process snoozes before rules as it might create a regression
             # but not if it's new because you can't immediately snooze a new group
-            has_reappeared = not is_new
-            try:
-                if has_reappeared:
+            if not is_reprocessed and has_reappeared:
+                try:
                     has_reappeared = process_snoozes(event.group)
-            except Exception:
-                logger.exception("Failed to process snoozes for group")
+                except Exception:
+                    logger.exception("Failed to process snoozes for group")
```

Comment:
Doesn't necessarily have to be in this pr, but I think ideally we'd have each function be self contained. So we'd just call `process_snoozes(event.group, is_reprocessed, is_new, ...)` and the logic for whether to run it could be decide in function, as well as the try/except. That'll make it easier to treat this as a pipeline.

Reply:
yeah that was going to be next PR. but got it